Add efpt instructions (#957)
gordon-cs/gordon-360-api: docs/Database.md

- `admintrainsql.gordon.edu` - The development/testing database server
- `adminprodsql.gordon.edu` - The live production database server.

### Updating context tables (_new columns, tables, procedures..._)

> SQL Table models are handled in `360api` by Entity Framework Power Tools (`efpt`). To script in new tables do the following:

Reviewer: This is really an oversimplification. Entity Framework Core is an ORM that handles mapping SQL tables/views/stored procedures to C# Objects, and transforming C# code into SQL Commands.

EF Core Power Tools is just a Visual Studio extension that adds a UI and some helpful reverse-engineering niceties on top of EF Core.

> SQL Table models are handled in `360api` by Entity Framework Power Tools (`efpt`). To script in new tables do the following:

Author: ```suggestion
Entity Framework 6 Power Tools are design-time utilities used in Visual Studio when working with the code-first development approach. As you know, the code-first approach does not support visual designer for the Entity Framework model. These power tools allow you to generate designer, XML and DDL SQL for your entities. In order to utilize this extension:
```

0) This _might_ need to be done: On to the top tab of Visual Studio
   - click `tools` -> `Connect to Database...`.
   - select `Microsoft SQL Server` -> `Continue`
   - <img width="509" alt="image" src="https://github.com/gordon-cs/gordon-360-api/assets/78386128/344d27eb-0882-44e6-812d-7ea4c7eb8be9">

     - select the appropriate Database Server (listed above)
     - select the appropriate database (CCT, MyGordon...)
     - press `ok`

>    - It is highly suggested to commit/stash your existing changes before doing an `efpt` edit.

Reviewer: This is not necessary, it's irrelevant to the EF Core connections.

Reviewer 2: Maybe say that it's helpful to work in a clean branch, so in step 2 it's easier to see what has changed and what might need to be reverted.

>    - It is highly suggested to commit/stash your existing changes before doing an `efpt` edit.

Author: ```suggestion
   - it is helpful to work on a clean branch/commit in order to clearly observe changes produced by `efpt`

```

    
1) Navigate on Solution Explorer to `\Models\{database name}\Context\`
   - <img width="243" alt="image" src="https://github.com/gordon-cs/gordon-360-api/assets/78386128/0a6b6ef8-9a5f-494e-b3b2-eeff298f1ccc">
   - right-click `efpt.{database name}.config.json`
   - select `EF Core Power Tools - edit`
   - select the appropriate Database. IF THERE ARE NO OPTIONS HERE, PLEASE RETURN TO STEP 0
   - press `ok`
   - select new database objects that you are trying to add
   - press `ok` after you're done selecting, then press `ok` on the next window
   - <img width="412" alt="image" src="https://github.com/gordon-cs/gordon-360-api/assets/78386128/a98b3d19-520a-4273-8a6f-6cdbfb50f6ff">

2) At this point, Git Changes should reflect changes to the API codebase of which there should be expected and unexpected changes.

   - <img width="275" alt="image" src="https://github.com/gordon-cs/gordon-360-api/assets/78386128/5d60e742-9496-424b-8a7e-674c51281a74">
    
   - At this point is is very important to note the changes that you expect and to revert all unexpected changes

>    - This is in exception to:

Reviewer 2: ```suggestion
   - Only keep these changes:
```

- `admintrainsql.gordon.edu` - The development/testing database server
- `adminprodsql.gordon.edu` - The live production database server.

### Updating context tables (_new columns, tables, procedures..._)

> Entity Framework 6 Power Tools are design-time utilities used in Visual Studio when working with the code-first development approach. As you know, the code-first approach does not support visual designer for the Entity Framework model. These power tools allow you to generate designer, XML and DDL SQL for your entities. In order to utilize this extension:

Reviewer 2: This says both more and less than Evan suggested. 

While there's obviously much more to say, there's enough here to help someone search for more online, and it's the instructions below that I think are most critical.
